ddtrace/tracer/tracer.go

```diff
@@ -371,3 +420,258 @@ func (t *tracer) sample(span *span) {
 	}
 	t.prioritySampling.apply(span)
 }
+
+type testStatsdClient struct {
+	mu          sync.RWMutex
+	gaugeCalls  []gaugeCall
+	incrCalls   []incrCall
+	countCalls  []countCall
+	timingCalls []timingCall
+	counts      map[string]int64
+	tags        []string
+	waitCh      chan struct{}
+	n           int
+	closed      bool
+}
+
+type gaugeCall struct {
```

Comment:
This is the same structure repeated 4 times. Why can't we just have one of them? `statsdCall`?